libsolidity/ExpressionCompiler.cpp

```diff
@@ -843,8 +843,11 @@ void ExpressionCompiler::endVisit(Identifier const& _identifier)
 	}
 	else if (FunctionDefinition const* functionDef = dynamic_cast<FunctionDefinition const*>(declaration))
 		m_context << m_context.getVirtualFunctionEntryLabel(*functionDef).pushTag();
-	else if (dynamic_cast<VariableDeclaration const*>(declaration))
-		setLValueFromDeclaration(*declaration, _identifier);
+	else if (auto variable = dynamic_cast<VariableDeclaration const*>(declaration))
+	{
+		if (!variable->isConstant())
```

Comment:
sure I don't forget it, I just didn't push last changes

On Thu, Mar 12, 2015 at 3:29 PM, chriseth <notifications@github.com> wrote:

> In libsolidity/ExpressionCompiler.cpp
> <https://github.com/ethereum/cpp-ethereum/pull/1274#discussion_r26305409>:
>
> > @@ -843,8 +843,11 @@ void ExpressionCompiler::endVisit(Identifier const& _identifier)
> >  	}
> >  	else if (FunctionDefinition const* functionDef = dynamic_cast<FunctionDefinition const*>(declaration))
> >  		m_context << m_context.getVirtualFunctionEntryLabel(*functionDef).pushTag();
> > -	else if (dynamic_cast<VariableDeclaration const*>(declaration))
> > -		setLValueFromDeclaration(*declaration, _identifier);
> > +	else if (auto variable = dynamic_cast<VariableDeclaration const*>(declaration))
> > +	{
> > +		if (!variable->isConstant())
>
> else variable->getValue()->accept(*this);
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ethereum/cpp-ethereum/pull/1274/files#r26305409>.
>